modules/authentication.py

```diff
@@ -169,11 +170,11 @@ def verify_response(
             session['macaroon_discharge']
         )
 
-        return {
-            'status_code': 307,
-            'redirect': url_calling,
-            'reason': 'Macaroon discharge refreshed'
-        }
+        raise ApiVerifyResponseError(
```

Comment:
I don't think checking if the macaroon needs refreshing should happen in the same function that we check for other general errors. And I'm also not sure that a Exception is the best way to communicate this state, or things like URLs. I haven't understood this code well enough to provide concrete suggestions, but I think probably we should have a "check authorization" step somewhere which is where things like macaroon freshness should be checked.